spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplication.java

```diff
@@ -92,12 +96,15 @@ private static void createChildContexts(ConfigurableApplicationContext parentCon
 			Class<?>[] apps, String args[][]) {
 		for (int i = apps.length - 1; i >= 0; i--) {
 			String appClassName = apps[i].getName();
-			embedApp(parentContext, getNamespace(appClassName, i), apps[i])
+			embedApp(parentContext, getNamespace(null, appClassName, i), apps[i])
 					.run(args != null ? args[i] : new String[0]);
 		}
 	}
 
-	protected static String getNamespace(String appClassName, int index) {
+	protected static String getNamespace(String namespace, String appClassName, int index) {
```

Comment:
Instead of passing `namespace` as a parameter here, we can use this method only when the appConfigurer doesn't have `namespace` set already.